java/vector/src/main/java/org/apache/arrow/vector/BufferLayout.java

```diff
@@ -70,7 +70,7 @@ public static BufferLayout dataBuffer(int typeBitWidth) {
       case 128:
         return VALUES_128;
       default:
-        throw new IllegalArgumentException("only 8, 16, 32, or 64 bits supported");
+        return new BufferLayout(BufferType.DATA, typeBitWidth);
```

Comment:
I think function is really "get data buffer layout for types that support certain bit widths" - so it might be beneficial to keep the exception, although it looks like it should be updated to include 128.  I'm not sure how exactly the exception would be thrown, but if somehow a buffer layout for an `Int` with 30 bit width was created, this would catch it.

Reply:
reverted and added 128 bits to the exception message